editor/src/components/editor/store/dispatch-strategies.tsx

```diff
@@ -406,50 +411,68 @@ function handleAccumulatingKeypresses(
   const canvasState: InteractionCanvasState = pickCanvasStateFromEditorState(newEditorState)
   // If there is a current strategy, produce the commands from it.
   if (newEditorState.canvas.interactionSession != null) {
-    const strategyResult =
-      strategy != null
-        ? applyCanvasStrategy(
-            strategy.strategy,
-            canvasState,
-            newEditorState.canvas.interactionSession,
-            strategyState,
-          )
-        : {
-            commands: [],
-            customState: strategyState.customStrategyState,
-          }
-    const commandResult = foldAndApplyCommands(
-      newEditorState,
-      storedEditorState,
-      strategyState.accumulatedPatches,
-      strategyState.currentStrategyCommands,
-      strategyResult.commands,
-      'transient',
-    )
-    const newStrategyState: StrategyState = {
-      currentStrategy: strategy?.strategy.id ?? null,
-      currentStrategyFitness: strategy?.fitness ?? 0,
-      currentStrategyCommands: strategyResult.commands,
-      accumulatedPatches: commandResult.accumulatedPatches,
-      commandDescriptions: commandResult.commandDescriptions,
-      sortedApplicableStrategies: sortedApplicableStrategies,
-      startingMetadata: strategyState.startingMetadata,
-      customStrategyState: strategyResult.customState ?? strategyState.customStrategyState,
-      startingAllElementProps: strategyState.startingAllElementProps,
-    }
+    const interactionData = newEditorState.canvas.interactionSession.interactionData
+    if (isKeyboardInteractionData(interactionData)) {
+      const lastKeyState = last(interactionData.keyStates)
+      const updatedInteractionData: KeyboardInteractionData = {
+        ...interactionData,
+        keyStates: lastKeyState == null ? [] : [lastKeyState],
+      }
```

Comment:
I have ambigous feelings about this. Until now the data flow looked like this:
- user interactions changed the interaction state
- based on the editor state and the interaction state strategies modified the state, but mostly the projectcontents itself and/or the StrategyState

But in this code the strategy handling modifies the interaction state itself, so the interaction state is not purely a description of the user behavior. I believe when the strategies need state, that should be in the StrategyState not in the InteractionState. So how I would do it is to put this information into the StrategyState, or to be more precise, probably into the StrategyState.customStrategyState.

Reply:
Surely `interactionDataHardReset` would fall into that as well? As that is triggered from `interactionHardReset` and it modifies the `drag` and `prevDrag` values.

Reply:
Good argument! It seems the separation between interaction and strategy states is just not that clear

Reply:
I'd definitely not put it in the `customStrategyState`, that should be reserved for things that are set by apply functions.